modules/nextflow/src/main/groovy/nextflow/processor/TaskConfig.groovy

```diff
@@ -377,6 +377,17 @@ class TaskConfig extends LazyMap implements Cloneable {
         return get('container')
     }
 
+    Architecture getArchitecture() {
+        final value = get('arch')
+        if( value instanceof String )
+            return new Architecture(value)
```

Comment:
```suggestion
        if( value instanceof CharSequence )
            return new Architecture(value.toString())
```

Otherwise `GString` (groovy strings) are not matched

Reply:
ok, I trust you in full...done!